bot/cogs/models/waifulistapi.py

```diff
@@ -262,20 +263,22 @@ async def fetch_random_waifu(self) -> Waifu:
         tree = etree.parse(StringIO(r.text), PARSER)
         waifu_id = tree.xpath("//waifu-core")[0].attrib[':waifu-id']
         waifu = await self.fetch_waifu_by_id(waifu_id)
-        self.cache[waifu_id] = waifu
         return waifu
 
     async def fetch_waifu_by_id(self, id: int) -> Waifu:
         """Returns a waifu by the specified ID."""
-        if self.cache.get(id):
-            return self.cache.get(id)
+        if self.waifu_cache.get(id):
+            return self.waifu_cache.get(id)
+
         r = await self.get(f"/waifu/{id}")
         waifu = Waifu(r.json()["data"])
-        self.cache[id] = waifu
+        self.waifu_cache[id] = waifu
         return waifu
 
     async def search(self, query: str) -> List[Union[WaifuSearchResult, SeriesSearchResult, SearchResult]]:
         """Returns a list of results matching your query."""
+        if self.search_cache.get(query):
```

Comment:
The key used in queries could be lowercased as I doubt it provides any difference in results